### ISSUE TITLE
Images don't show on Google searches with v1.7.6: RESOLVED.

### ISSUE THREAD

**Issue by rumorumo** [gorhill/uBlock#1782]
Chrome 31.0.1650.63 (on Windows XP) and uBlock 1.7.6 (or 1.7.7).

My filters (+ default filters):
```
www.ig.com.br###h_banner
www.ig.com.br###BB3539DF-AF6E-4A91-A1BE-54E0E0E8D64F
www.ig.com.br###E655119E-64A4-4943-BBD2-404A838A2B6B
www.ig.com.br###\31 7E59153-3A9D-4672-95C0-E2BD0246BDC2
```

My rules:
```
* * 3p-frame block
* * 3p-script block
www.ig.com.br ajax.googleapis.com * noop
www.ig.com.br http://www.googletagmanager.com/gtm.js?id=GTM-TTKT6C script allow
www.ig.com.br js.statig.com.br * noop
www.ig.com.br maps.googleapis.com * noop
```

URL: http://www.ig.com.br

With uBlock 1.7.4 or earlier it's OK. From 1.7.6 on most pictures don't show (even in easy mode), and trying new rules has no effect (only disabling uBlock altogether makes a difference).

Thank you!

**Comment by gorhill** [comment]
Unable to reproduce on Chromium 51 that is. I do not have the resource to test on very old version of Chromium. Somebody else will have to fork and fix it if you think there is an issue with uBO proper.

**Comment by rumorumo** [comment]
Well, until 1.7.4 the issue was definitely not there. Something changed with 1.7.6, but I suspect it's hardly going to be fixed by anyone else, since it was labeled "declined" and "closed"... Thanks anyway.

**Comment by rumorumo** [comment]
Same problem with Google searches...

With uBlock v1.7.4:
![v174](https://cloud.githubusercontent.com/assets/15052951/16672820/0c4c935e-447f-11e6-9f59-42e5ef9de2ce.JPG)

With v1.7.6:
![v176](https://cloud.githubusercontent.com/assets/15052951/16672867/504fe31c-447f-11e6-978b-5ff68a2017a4.JPG)

Again with v1.7.4:
![v174a](https://cloud.githubusercontent.com/assets/15052951/16672878/644ae466-447f-11e6-9ec6-5213126c2429.JPG)

Again with v1.7.6:
![v176a](https://cloud.githubusercontent.com/assets/15052951/16672906/9652e2d8-447f-11e6-8482-dbfeb19da787.JPG)

**Comment by gorhill** [comment]
Did you investigate with the logger? You did not provide any ready to use URLs from your screenshots.

**Comment by rumorumo** [comment]
Yes, I have done everything I know with the logger, even allowing all that was being blocked, and the problem remained. It seems that some code change introduced in uBlock 1.7.6 has affected the way the old Chrome 31 works with these images. I know it's an old version, but who knows if the problem will not crop up anytime with a new version?

Sorry for the URLs. Any Google search for images shows the problem here (with the old Crome 31, and only from uBlock 1.7.6 on). There goes one:

https://www.google.com/search?q=screenshots+don't+scare+me&biw=1280&bih=908&tbm=isch&tbo=u&source=univ&sa=X&sqi=2&ved=0ahUKEwiWyqamxuLNAhWHPpAKHSuDDUsQsAQIGw

**Comment by gorhill** [comment]
I still can't reproduce with your URL.

An observation however: for the first three rows of images, the `src` attribute of all those images starts with `data:...`, i.e. it is a data URI. For the last (4th) row, the `src` attribute is a plain `https://...` one.

There has been a change in 1.7.6, which is that `data:...` images are now allowed to be filtered through the network filtering engine. However, I fail to see how you could get the result you get -- as said, they are properly displayed on my side. You have custom filters?

**Comment by rumorumo** [comment]
I have also tested both 1.7.6 and 1.7.7 out of the box, without any custom filters and rules, and the problem persists. I could observe that each time I load the pages the missing images show initially for less than one second and then disappear. Testing again moments ago, I get the following strange results with 1.7.7b10, *but only after updating the default filters*. If there's something else I could try for helping, just ask.

![v177b](https://cloud.githubusercontent.com/assets/15052951/16693213/8d5d957a-450a-11e6-8c2d-59b6550898c6.JPG)

![v177c](https://cloud.githubusercontent.com/assets/15052951/16693221/967358c0-450a-11e6-8557-97a26cc256b8.JPG)

**Comment by gorhill** [comment]
Maybe there is javascript code that execute differently in Chromium 31. Problem is that I do not have access to such an old version (I am on linux) -- so I can't investigate.

**Comment by T24G** [comment]
Exact same thing is happening to me in Opera 25.0.1614.71 with uBlock 1.7.6. Disabling uB prevents it from happening, but reenabling immediately causes the issue again. I get the same colored blocks instead of images on both the main search page, as well as the images page, just like @rumorumo.
![screen shot 2016-07-12 at 5 15 10 pm](https://cloud.githubusercontent.com/assets/1143397/16783928/66eff9bc-4854-11e6-9212-1ee669ab1686.jpg)

**Comment by TheRyuu** [comment]
> Maybe there is javascript code that execute differently in Chromium 31. Problem is that I do not have access to such an old version (I am on linux) -- so I can't investigate.

There is no LTS/ESR type release channel for Chrome so I wouldn't be surprised if it was near impossible to get such an old version of Chrome.  It should go without saying that a Chrome version that old has any number of unpatched security issues.

**Comment by gorhill** [comment]
In light of [this issue](https://github.com/uBlockOrigin/uAssets/issues/85), could you apply the same fix to see if this helps with the issue here?

**Comment by 123466769999Ram** [comment]
# I think i've found the  solution. 
The main problem was with EasyList and EasyList associated filters. They do not work properly sometimes, or to be fair, most of the times.  This is my settings, I remove every EasyList filter and some filters that are directing you to "forums.lanik.us" which is a EasyList assocaited forum. So this is my seetings: 
![screenshot_2](https://cloud.githubusercontent.com/assets/20622871/17083038/3cd8c956-519b-11e6-9080-08f5b1d77baa.png)
![screenshot_4](https://cloud.githubusercontent.com/assets/20622871/17083039/3cd99d90-519b-11e6-91a6-feed0d75b37e.png)



[ublock-backup_24.7.2016_12-14-14.txt](https://github.com/uBlockOrigin/uAssets/files/380212/ublock-backup_24.7.2016_12-14-14.txt)

Here is proof:
![screenshot_5](https://cloud.githubusercontent.com/assets/20622871/17083062/fd07ce02-519b-11e6-8c1c-f6db510ac21a.png)

I hope that helped you, with this settings, where EasyList's filters and "forums.lanik.us" are disabled, everything with Google pictures should be fine.  Good Luck.

**Comment by rumorumo** [comment]
@gorhill I've experienced the problem updating EasyList and EasyPrivacy, but only with 1.7.6 and 1.7.7 (never with 1.7.4). This specific issue was solved with your suggested fix (changing the URLs), but it didn't helped with the images problem. Unselecting EasyList, as proposed above, definitely made it, but I believe this solution is far from ideal (and anyway with 1.7.4 it's absolutely unnecessary to do so). Which changes in 1.7.6 could affect the way EasyList is used?

**Comment by gorhill** [comment]
> Unselecting EasyList, as proposed above, definitely made it

Si this means the logger should show what is happening. Compare the logger output with and without EasyList. Report the difference.

**Comment by rumorumo** [comment]
Here is it. First with EasyList (and no images):

![176witheasylist](https://cloud.githubusercontent.com/assets/15052951/17086435/5ec0cce4-51c8-11e6-9185-531cee1a3894.JPG)

And without it (images OK):

![176withouteasylist](https://cloud.githubusercontent.com/assets/15052951/17086456/f0778aec-51c8-11e6-8a30-b5d9d97004bd.JPG)

**Comment by gorhill** [comment]
It's a "blind" fix: I can't reproduce, thus I can't verify whether the fix works, this needs confirmation. In any case, I think treating `data:` URI images as 1st-party resources is reasonable, since these resources are delivered inlined in the document.

**Comment by rumorumo** [comment]
Apparently the fix didn't work. The logger output didn't change (with and without EasyList):

![177witheasylist](https://cloud.githubusercontent.com/assets/15052951/17103194/9a46e51a-5253-11e6-90e8-6a88a3fe0b2d.JPG)

**Comment by gorhill** [comment]
Forget the logger, `data:` URIs do not appear in the logger, as they are not going through the webRequest API. The only way to figure out what is going on on your side is to step into the code to see why the image with a `data:` URI source are being collapsed -- since I can't reproduce on my side, someone will have to step up to investigate this issue.

**Comment by kasper93** [comment]
I can reproduce the issue on http://www.ig.com.br/ and actually it is the http://www.googletagmanager.com/gtm.js script.  If block it website will not load anymore scripts. Also "Experimental" with redirect to "noop" version doesn't help here. 

If you still can't reproduce I can help more if needed.

**Comment by rumorumo** [comment]
I have noticed an erratic behavior of the missing images, a little different from the pattern I have already referred to. As I have said, they all appear for less than a second and go away. But at times and randomly just some disappear while most of them (once in a while even all) remain there. As always, nothing wrong like this occur with 1.7.4 or with EasyList unselected.

**Comment by gorhill** [comment]
With fixing #1927, I removed the code I had put for uBO to allow `data:` URI to be handled by the static filtering engine -- which was to fix #1725, but for which in the end I chose a different fix.

So could you see if you still have the issue with 1.8.5rc1?

**Comment by rumorumo** [comment]
The issue is gone. Thank you again.
